src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -2185,3 +2283,44 @@ function isSuitableCategoricalVariable(variable?: Variable): boolean {
     variable.distinctValuesCount != null
   );
 }
+
+/**
+ *  A hook to handle default dependent axis range for Lineplot Viz Proportion
+ */
+function useDefaultDependentAxisRangeProportion(
+  data: PromiseHookState<LinePlotDataWithCoverage | undefined>,
+  yAxisVariable?: Variable,
+  dependentAxisLogScale?: boolean,
+  valueSpecConfig?: string,
+  dependentAxisValueSpec?: string
+) {
+  let defaultDependentAxisRange = useDefaultAxisRange(
+    yAxisVariable,
+    data.value?.yMin,
+    data.value?.yMinPos,
+    data.value?.yMax,
+    dependentAxisLogScale,
+    dependentAxisValueSpec
+  );
+
+  // include min origin: 0 (linear) or minPos (logscale)
+  if (data.value != null && valueSpecConfig === 'Proportion')
+    if (dependentAxisLogScale)
+      defaultDependentAxisRange = {
+        min: data.value?.yMinPos,
+        // in case data.value.yMinPos === 1, then use 1.0001 for max range for better display
```

Comment:
Does the 1.0001 avoid points being chopped off at the top? Don't we add padding already, somewhere else?

Reply:
@bobular For Lineplot proportion, there is a case that both min and max are unity (1) (e.g., numerator = denominator, etc.), then the range becomes [1,1]. It plots okay without error but the y-axis tick labels become strange - so it is just for a better display purpose - you can see it without setting 1.0001 😃.

Reply:
Oh, I see.  I mis-read line 2305 as checking the max, not min!

I have just been looking at this, and comparing with barplot and histogram.  Can they all use the same logic to handle proportion mode?  I see that barplot and histogram both have a bit of [cut-and-pasted similar code](https://github.com/VEuPathDB/web-eda/blob/5065d207a7e636a3b140eb1e17c944c7052b8f0c/src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx#L477-L488) - could we add a `isProportion : boolean` prop to useDefaultAxisRange and handle everything in there?  It would be better for future maintainability.

Maybe we could deal with the `min === max` edge case in `web-components` extend-axis-range-truncations.ts?

Reply:
> I have just been looking at this, and comparing with barplot and histogram. Can they all use the same logic to handle proportion mode? I see that barplot and histogram both have a bit of [cut-and-pasted similar code](https://github.com/VEuPathDB/web-eda/blob/5065d207a7e636a3b140eb1e17c944c7052b8f0c/src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx#L477-L488) - could we add a `isProportion : boolean` prop to useDefaultAxisRange and handle everything in there? It would be better for future maintainability.

@bobular Actually I tried to make it be common but it was not that simple task because: a) there exist many conditions (log scale, dataType, dataRange, etc.) and it adds more complexity; b) I cannot remember correctly, but it caused type issue. Due to those reasons, I deferred it and made things work at first. Perhaps technical debt item.
 
> Maybe we could deal with the `min === max` edge case in `web-components` extend-axis-range-truncations.ts?

Yeah it may be better to handle those edge case in there. Another tech debt item? :)

Reply:
Great, let's get this merged relatively soon so we can get feedback on various functional aspects.

I've added a new [tech debt ticket here](https://github.com/VEuPathDB/web-eda/issues/1398).